internal/scaffold/role.go

```diff
@@ -143,6 +145,69 @@ func UpdateRoleForResource(r *Resource, absProjectPath string) error {
 	}
 }
 
+// MergeHelmRoleForResource merges incoming new API resource rules with existing deploy/role.yaml
+func MergeHelmRoleForResource(r *Resource, absProjectPath string, roleScaffold Role) error {
+	roleFilePath := filepath.Join(absProjectPath, DeployDir, RoleYamlFile)
+	roleYAML, err := ioutil.ReadFile(roleFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read role manifest %v: %v", roleFilePath, err)
+	}
+	obj, _, err := cgoscheme.Codecs.UniversalDeserializer().Decode(roleYAML, nil, nil)
+	if err != nil {
+		return fmt.Errorf("failed to decode role manifest %v: %v", roleFilePath, err)
+	}
+	switch role := obj.(type) {
+	case *rbacv1.Role:
+		for j := range roleScaffold.CustomRules {
+			prj := &roleScaffold.CustomRules[j]
+		iLoop:
+			for i := range role.Rules {
+				pri := &role.Rules[i]
+				// check if apiGroup, verbs, resourceName, and nonResourceURLS matches for new resource.
+				apiGroupsEqual := ifMatches(pri.APIGroups, prj.APIGroups)
+				verbsEqual := ifMatches(pri.Verbs, prj.Verbs)
+				resourceNamesEqual := ifMatches(pri.ResourceNames, prj.ResourceNames)
+				nonResourceURLsEqual := ifMatches(pri.NonResourceURLs, prj.NonResourceURLs)
+
+				if apiGroupsEqual && verbsEqual && resourceNamesEqual && nonResourceURLsEqual {
+					for _, newResource := range prj.Resources {
+						if !findResource(pri.Resources, newResource) {
+							// append rbac rule to deploy/role.yaml
+							pri.Resources = append(pri.Resources, newResource)
+						}
+					}
+					break iLoop
+				}
+			}
+		}
+		return updateRoleFile(&role, roleFilePath)
+
+	default:
+		return errors.New("failed to parse role.yaml as a role")
+	}
+}
+
+func ifMatches(pr []string, newPr []string) bool {
+	if len(pr) != len(newPr) {
+		return false
+	}
+	for i, v := range pr {
```

Comment:
This would return `false` is the lists were:
- [pr1, pr2]
- [pr2, pr1]

We should sort both lists before comparing using `sort.Strings()`

Reply:
This one totally makes sense, I have been seeing some difference in my unit tests due to not sorting, most likely.